### ISSUE TITLE
Rename core workflow

### ISSUE THREAD

**Issue by corverroos** [ObolNetwork/charon#534]
## Problem to be solved

We currently use the term "aggregate" in many names in the core workflow. This is however incorrect, since we are referring to threshold BLS signature "combination" (M-of-N multisig). "Aggregation" actually means normal BLs signature aggregation, as used in ETH2. 

We therefore need to rename all "agg" components.

We might also take this opportunity to improve naming of a few other entities as well,:
 -  `core.FetchArgs` was originally only used as input arguments to `fetcher`, so `FetchArgs` made sense. But this is now also used in `validatorapi`. Since `FetchArgs` define the "duty" to be performed, we are suggesting "DutyDefinition" as the new name.
  
**The following is a list of all OLD core workflow names, the groups they form, and contexts they are used in:**

---
 - `FetchArgs` - fetch arguments - duty definition, used to fetch duty data
 - `ParSignedData` - partial signature data - duty data containing partial bls signatures
 - `AggSignedData` - aggregated signature data - duty data containing aggregated bls signatures
 - `UnsignedData` - unsigned data - duty data to be signed
 ---
 - `FetchArgSet` - a set of fetch arguments
 - `ParSignedDataSet` - a set of partial signature data
 - `AggSignedDataSet` - a set of aggregated signature data
 - `UnsignedDataSet` - a set of unsigned data
 --- 
 - `ParSigDB` - partial signature database
 - `ParSigEx` - partial signature exchange
 - `SigAgg` - signature aggregator
 - `AggSigDB` - aggregated signature database
 ---
 Concatenated method names:
 -  `Decode*` - e.g. `DecodeAggSignedDataSet`
 -  `Encode*` - e.g. `EcnodeUnsignedDataSet`
 - `*ToProto` - e.g `ParSignedDataToProto`
 - `*FromProto` - e.g. `AggSignedDataFromProto`
 ---
 
## Proposed solution

The suggested renaming approach is:
 - Partial signatures and data signed with them get a `Share` prefix.
 - t-BLS "Combined" signatures and data signed with them get a `Group` prefix.
 - So `shares` are "combined" into a `group`.
 
**The following is a list of all NEW core workflow names, the groups they form, and contexts they are used in:**

---
 - `FetchArgs` -> `DutyDefinition` - duty definition, used to fetch duty data
 - `ParSignedData` -> `ShareSignedData`- partial signatures and data containing partial signatures 
 - `AggSignedData` -> `GroupSignedData`- combined signatures and data containing combined signatures 
 - `UnsignedData` - (don't rename) - unsigned data not containing signatures 
 ---
 - `FetchArgSet` -> `DutyDefinitionSet` - a set of fetch arguments
 - `ParSignedDataSet`-> `ShareSignedDataSet`  - a set of partial signature data
 - `AggSignedDataSet` > `GroupSignedDataSet`  - a set of combined signature data
 - `UnsignedDataSet` - (don't rename) - a set of unsigned data
 --- 
 - `ParSigDB` -> `ShareSigDB`- partial signature database
 - `ParSigEx`-> `ShareSigExchange` - partial signature exchange
 - `SigAgg` -> `Combiner` - partial signatures combiner
 - `AggSigDB` -> `GroupSigDB`- combined signature database
 ---
 Concatenated method names:
 -  `Decode*` - e.g. `DecodeGroupSignedDataSet`
 -  `Encode*` - e.g. `EncodeUnsignedDataSet`
 - `*ToProto` - e.g `ShareSignedDataSetToProto`
 - `*FromProto` - e.g. `DutyDefinitionSetFromProto`
 ---

All docs and code and tests should be renamed accordingly.

**Comment by leolara** [comment]
In general I like these names better. I would do `Combiner` -> `BLSCombiner` or something like that, that is not so general.

**Comment by leolara** [comment]
Or `SigCombiner`?

**Comment by dB2510** [comment]
I don't think we need to rename core workflow. Because we are doing threshold BLS Signature Aggregation not combination. I think the confusion has been caused by the function naming on kryptology which is CombineSignatures which is essentially doing Threshold Aggregation. I think this is on kryptology not on us.

I have also tried to compare the 2 types of aggregation here: https://db2510.github.io/blogs/aggregation/
